examples/min64/export-image/04-finalise/01-run.sh

```diff
@@ -0,0 +1,118 @@
+#!/bin/bash -e
+
+IMG_FILE="${STAGE_WORK_DIR}/${IMG_FILENAME}${IMG_SUFFIX}.img"
+INFO_FILE="${STAGE_WORK_DIR}/${IMG_FILENAME}${IMG_SUFFIX}.info"
+
+sed -i 's/^update_initramfs=.*/update_initramfs=all/' "${ROOTFS_DIR}/etc/initramfs-tools/update-initramfs.conf"
+
+on_chroot << EOF
+update-initramfs -k all -c
+if [ -x /etc/init.d/fake-hwclock ]; then
+	/etc/init.d/fake-hwclock stop
+fi
+if hash hardlink 2>/dev/null; then
+	hardlink -t /usr/share/doc
+fi
+EOF
+
+if [ -d "${ROOTFS_DIR}/home/${FIRST_USER_NAME}/.config" ]; then
+	chmod 700 "${ROOTFS_DIR}/home/${FIRST_USER_NAME}/.config"
+fi
+
+rm -f "${ROOTFS_DIR}/usr/bin/qemu-arm-static"
+
+if [ "${USE_QEMU}" != "1" ]; then
+	if [ -e "${ROOTFS_DIR}/etc/ld.so.preload.disabled" ]; then
+		mv "${ROOTFS_DIR}/etc/ld.so.preload.disabled" "${ROOTFS_DIR}/etc/ld.so.preload"
+	fi
+fi
+
+rm -f "${ROOTFS_DIR}/etc/network/interfaces.dpkg-old"
```

Comment:
I'm not super-keen on this mechanism, because you wind up with 'export image' performing intrusive modifications on the system that may produce effects that you'll only see in later QA.

I'd suggest exploring the mechanism we deployed in pi-gen-micro: https://github.com/raspberrypi/pi-gen-micro/blob/main/configurations/cryptroot_image/dpkg_extra_args

Essentially, that mechanism prevents files matching a globbing pattern you specify from _ever being installed_. This is handy, because it means you can also have correctness checks as part of your regular build flow. Performing the modifications here doesn't really allow for that without re-mounting the image, or running a second step after pi-gen.

Reply:
> I'm not super-keen on this mechanism, because you wind up with 'export image' performing intrusive modifications on the system that may produce effects that you'll only see in later QA.
> 
> I'd suggest exploring the mechanism we deployed in pi-gen-micro: https://github.com/raspberrypi/pi-gen-micro/blob/main/configurations/cryptroot_image/dpkg_extra_args

I just copied what we currently do today, but I agree.